backend/btrixcloud/crawlconfigs.py

```diff
@@ -1052,6 +1080,94 @@ async def re_add_all_scheduled_cron_jobs(self):
                     flush=True,
                 )
 
+    def _validate_behavior_git_repo(self, repo_url: str, branch: str = ""):
+        """Validate git repository and branch, if specified, exist and are reachable"""
+        git_remote_cmd = ["git", "ls-remote", repo_url, "HEAD"]
+        try:
+            subprocess.run(git_remote_cmd, check=True)
```

Comment:
I think we probably want to use the asyncio subprocess API here, eg:
https://docs.python.org/3/library/asyncio-subprocess.html

Should be fairly easy to switch, since we just want the exit code.